com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/OutlineServiceTest.java

```diff
@@ -0,0 +1,82 @@
+package com.ca.lsp.cobol.service;
+
+import com.ca.lsp.cobol.service.delegates.validations.AnalysisResult;
+import com.ca.lsp.core.cobol.semantics.outline.*;
+import lombok.Value;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class OutlineServiceTest {
+  @Test
+  public void collectSymbols() {
+    MyDocumentModel documentModel = new MyDocumentModel("",
+        new AnalysisResult(null, null, null, null, null, null, null,
+            new OutlineTree(List.of(outlineNode("FOO", NodeType.FIELD, from(3, 3), to(4, 4))))));
+    List<Either<SymbolInformation, DocumentSymbol>> expectedResult = List.of(Either.forRight(
+        documentSymbol("FOO", SymbolKind.Variable, from(2, 3), to(3, 4))
+    ));
+    assertEquals(expectedResult, OutlineService.collectSymbols(documentModel));
+  }
+
+  @Test
+  public void collectSymbolsForNullDocumentModel() {
+    assertEquals(Collections.EMPTY_LIST, OutlineService.collectSymbols(null));
+  }
+
+  @Test
+  public void collectSymbolsForNullOutlineTree() {
+    MyDocumentModel documentModel = new MyDocumentModel("",
+        new AnalysisResult(null, null, null, null, null, null, null, null));
+    assertEquals(Collections.EMPTY_LIST, OutlineService.collectSymbols(documentModel));
+  }
+
+  @Test
+  public void convert() {
+    // The structure is the following:
+    // root
+    // |- first
+    // |  |- FILLER
+    // |- second
+    OutlineNode outlineNode = outlineNode("root", NodeType.PROGRAM, from(1, 1), to(10, 45),
+        outlineNode("first", NodeType.DIVISION, from(2, 2), to(2, 10),
+            outlineNode("FILLER", NodeType.STRUCT, from(3, 4), to(3, 8))),
+        outlineNode("second", NodeType.COPYBOOK, from(5, 3), to(5, 20)));
+    DocumentSymbol expectedDocumentSymbol = documentSymbol("root", SymbolKind.Class, from(0, 1), to(9, 45),
+        documentSymbol("first", SymbolKind.Method, from(1, 2), to(1, 10),
+            documentSymbol("FILLER", SymbolKind.Struct, from(2, 4), to(2, 8))),
+        documentSymbol("second", SymbolKind.File, from(4, 3), to(4, 20)));
```

Comment:
might be better to wrap it in a method to improve readability

Reply:
I decided to keep them in one place because it easy to understand how the outline structure must be converted into a document symbol. Like: "first" DIVISION from outline node became "first" Method as a document symbol.
Or from(2, 2) for the "first" must become from(1, 2).
If I split the code by methods it will be hard to visually compare these two formats.

Reply:
👍